napari/layers/labels/_tests/test_labels.py

```diff
@@ -349,6 +349,42 @@ def test_n_dimensional():
     assert layer.n_dimensional is True
 
 
+def test_contour():
+    """Test changing contour."""
+    data = np.array(
+        [
+            [1, 1, 0, 0, 0, 1, 1, 1, 2, 1],
+            [2, 1, 0, 0, 0, 0, 1, 1, 1, 0],
+            [2, 2, 2, 0, 2, 2, 1, 2, 2, 0],
+            [1, 2, 0, 0, 2, 1, 2, 0, 0, 0],
+            [1, 0, 0, 0, 1, 2, 1, 2, 1, 2],
+        ]
+    )
+    layer = Labels(data)
+    assert layer.contour is False
+    np.testing.assert_array_equal(layer.data, data)
+
+    layer.contour = True
+    assert layer.contour is True
+
+    # Check `layer.data` didn't change
+    np.testing.assert_array_equal(layer.data, data)
+
+    # Check what is returned in the view of the data
+    np.testing.assert_array_equal(
+        1 * layer._raw_to_displayed(data),
```

Comment:
Trying to test this... is there a better way to check what's inside the view of the data after setting the `contour` property?

Reply:
I'm not 100% sure, but maybe `layer._data_view` would work. Perhaps @sofroniewn has a better idea.

https://github.com/napari/napari/blob/2a6adfa4fdc58be96140b04774ae2c7ddc1701b0/napari/layers/image/image.py#L305-L307

Reply:
Yes, using `layer._data_view` here would be good too

Reply:
If the `refresh` stuff is all working then setting `layer.contour=True` should update `layer._data_view`

Reply:
It seems to work as you both said :+1: